### ISSUE TITLE
5.0.1 - Erroneous 'warn' log message when --config.debug is used

### ISSUE THREAD

**Issue by patrickpilch** [elastic/logstash#6256]
When `--log.level=debug` and `--config.debug` are provided, a `WARN` level log message is produced incorrectly stating the log level was not set to 'debug'.

- Version: **5.0.1**
- Operating System: **OS X Darwin Kernel Version 15.6.0**
- Config File:
```ruby
input { stdin { } }
output { stdout { codec => json } }
```
- Sample Data: **Anything**
- Steps to Reproduce:
```bash
$ ./logstash -f ./test.conf --log.level=debug --config.debug
```
```
...

[2016-11-16T11:01:43,507][WARN ][logstash.runner          ] --config.debug was specified, but log.level was not set to 'debug'! No config info will be logged.

...

[2016-11-16T11:01:43,699][DEBUG][logstash.agent           ] 
input { stdin {} }
output { stdout { codec => json } }



[2016-11-16T11:01:43,700][DEBUG][logstash.agent           ] 
The following is the merged configuration
[2016-11-16T11:01:43,700][DEBUG][logstash.agent           ] 
input { stdin {} }
output { stdout { codec => json } }

```

**Comment by jsvd** [comment]
this is still true in the latest snapshot:

```
% ./bin/logstash -f /tmp/cfg --log.level=debug --config.debug
Sending Logstash's logs to /tmp/logstash-5.2.0/logs which is now configured via log4j2.properties
[2017-01-06T18:08:04,709][WARN ][logstash.runner          ] --config.debug was specified, but log.level was not set to 'debug'! No config info will be logged.
```

**Comment by suyograo** [comment]
Fixed in #6789
